site/content/extend/plugins/webapp/actions.md

```diff
@@ -1,115 +1,493 @@
 
-This guide assumes you have already set up your plugin development environment for web app plugins to match [mattermost-plugin-starter-template](https://github.com/mattermost/mattermost-plugin-starter-template). If not, follow the README instructions of that repository first, or [see the Hello, World! guide](/extend/plugins/webapp/hello-world/).
+This guide assumes you've already set up your plugin development environment for web app plugins to match [mattermost-plugin-starter-template](https://github.com/mattermost/mattermost-plugin-starter-template). If not, follow the README instructions of that repository first, or [see the Hello, World! guide](/extend/plugins/webapp/hello-world/).
 
-## Import mattermost-redux
+## Basic example
 
-First, you'll need to add `mattermost-redux` as a dependency of your web app plugin.
+Here's an [example](https://github.com/mattermost/mattermost-plugin-jira/blob/master/webapp/src/components/modals/create_issue/index.ts) of a web app plugin making use of [mattermost-redux's](https://github.com/mattermost/mattermost-redux) functions:
 
-```bash
-cd /path/to/plugin/webapp
-npm install mattermost-redux
+```javascript
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+
+import {closeCreateModal, createIssue, fetchJiraIssueMetadataForProjects, redirectConnect} from 'actions';
+import {isCreateModalVisible, getCreateModal} from 'selectors';
+
+import CreateIssue from './create_issue_modal';
+
+const mapStateToProps = (state) => {
+    const {postId, description, channelId} = getCreateModal(state);
+    const post = (postId) ? getPost(state, postId) : null;
+    const currentTeam = getCurrentTeam(state);
+
+    return {
+        visible: isCreateModalVisible(state),
+        post,
+        description,
+        channelId,
+        currentTeam,
+    };
+};
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+    close: closeCreateModal,
+    create: createIssue,
+    fetchJiraIssueMetadataForProjects,
+    redirectConnect,
+}, dispatch);
+
+export default connect(mapStateToProps, mapDispatchToProps)(CreateIssue);
 ```
 
-That will add `mattermost-redux` as a dependency in your `package.json` file, allowing it to be imported into any of your plugin's JavaScript files.
+## Some common actions
 
-## Use an Action
+We've listed out some of the commonly-used actions that you can use in your web app plugin. You can find all the actions that are available for your plugin to import [in the source code for mattermost-redux](https://github.com/mattermost/mattermost-redux/tree/master/src/actions).
 
-Actions are used as part of components. To give components access to these actions, we pass them in as React props from the component's container `index.js` file. To demonstrate this, we'll create a new component.
+**createChannel**
```

Comment:
Yeah, we definitely can link. But, when the Dev is on this page, he should be able to know about the common ones that are listed here, and for something extra, he can always hop into the code base which is linked at the beginning of each section. Apart from that, I have applied the formatting so that they don't use up that much space on the page. Collapsible divs have been added for each of the sections. From a user perspective, keeping the snippet for the common ones here seems more user-friendly and adds more value to the page. I hope you will like this one.